plugins/colorsource/export.js

```diff
@@ -0,0 +1,283 @@
+// see https://github.com/standard-things/esm#getting-started
+require = require(`esm`)(module); // eslint-disable-line no-global-assign
+
+const {
+  CoarseChannel,
+  FineChannel,
+  SwitchingChannel
+} = require(`../../lib/model.js`);
+const { scaleDmxValue } = require(`../../lib/scale-dmx-values.mjs`);
+
+const CHANNEL_TYPE_NO_FUNCTION = 0;
+const CHANNEL_TYPE_INTENSITY = 1;
+const CHANNEL_TYPE_POSITION = 2;
+const CHANNEL_TYPE_COLOR_TEMP = 3;
+const CHANNEL_TYPE_BEAM = 4;
+const CHANNEL_TYPE_COLOR = 5;
+
+module.exports.name = `ColorSource`;
+module.exports.version = `0.1.0`;
+
+const EDITOR_VERSION = `1.1.1.9.0.4`;
+
+/**
+ * @param {array.<Fixture>} fixtures An array of Fixture objects.
+ * @param {object} options Global options, including:
+ * @param {string} options.baseDir Absolute path to OFL's root directory.
+ * @param {Date|null} options.date The current time.
+ * @returns {Promise.<array.<object>, Error>} The generated files.
+*/
+module.exports.export = function exportColorSource(fixtures, options) {
+  const exportJson = {
+    date: options.date.toISOString().replace(/\.\d\d\dZ$/, `Z`),
+    editorVersion: EDITOR_VERSION,
+    personalities: []
+  };
+
+  fixtures.forEach(fixture => {
+    fixture.modes.forEach(mode => {
+      const hasIntensity = mode.channels.some(ch => ch.type === `Intensity`);
+
+      const fixtureJson = {
+        colortable: null,
+        commands: getCommands(mode),
+        hasIntensity: hasIntensity,
+        manufacturerName: fixture.manufacturer.name,
+        maxOffset: mode.channels.length - 1,
+        modeName: mode.name,
+        modelName: fixture.name,
+        parameters: getCSChannels(mode, hasIntensity)
+      };
+
+      fixtureJson.colortable = getColorTable(fixtureJson.parameters);
```

Comment:
Better move `parameters` to separate variable above the object creation and use it in both `parameters` (directly) and `colortable` (with this function call) instead of overwriting the `null` value.